### PR TITLE
[ONNX] Modified slice node in inplace ops pass

### LINKED ISSUE #42292: bug in conversion torch model to onnx model
## 🐛 Bug
I'm converting a pytorch model to onnx model. in this model there assignment of tensor to a slice of another tensor.
when i'm running the converted model with onnxruntime he crashes when trying to assign the small tensor to the big tensor and ignoring the slice operation.

## To Reproduce
i created a stand alone model to reproduce the problem. this is the error when trying to run the model:

**The input tensor cannot be reshaped to the requested shape. Input shape:{22,256}, requested shape:{56,1,256}**

run the attached code

``import torch
import onnx
import onnxruntime

class Test(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
        x[:y.size(0), 0, :] = y
        return x

x = torch.zeros((56,6,256))
y = torch.rand((22,256))

m = Test()
torch_outputs = m(x, y)

path = "/tmp/m.onnx"
torch.onnx.export(m, (x, y), path,
                    do_constant_folding=True,
                    opset_version=12,
                    input_names=["x", "y"],
                    output_names=["z"])

onnx_model = onnx.load_model(path)
onnx.checker.check_model(onnx_model)
inferred_model = onnx.shape_inference.infer_shapes(onnx_model)
onnx.save(inferred_model, path)

ort_model = onnxruntime.InferenceSession(path)

ort_outputs = ort_model.run(["z"], {"x": x.numpy(), "y": y.numpy()})``

## Expected behavior
run successfully the model with onnxruntime

## Environment

 - PyTorch Version: 1.5.1 
 - OS (e.g., Linux): Linux ubuntu 16.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.6.8
 - GPU models and configuration:
 - Any other relevant information: onnxruntime verion 1.4.0

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Fixes #42292

### CODE REVIEW

**Comment by BowenBao** [comment]
LG, please fix the lint failure.

**Review by BowenBao — APPROVED** [review]
LGTM, please ensure CI is all green by rerun/rebase.

**Review by bzinodev — APPROVED** [review]

**Comment by shubhambhokare1** [comment]
@bzinodev build failure seems unrelated to changes made in code

File: torch/csrc/jit/passes/onnx/prepare_inplace_ops_for_onnx.cpp [hunk 1]
```diff
@@ -27,9 +27,10 @@ Value* ConvertSliceToIndex(Node* slice, Value* size, Node* insertBefore) {
   const int64_t int_max = std::numeric_limits<int>::max();
   auto graph = slice->owningGraph();
   WithInsertPoint guard(insertBefore);
-  auto start = slice->get(attr::start) ? slice->get(attr::start) : 0;
-  auto end = slice->get(attr::end) ? slice->get(attr::end) : int_max;
-  auto step = slice->get(attr::step);
+  TORCH_INTERNAL_ASSERT((slice->inputs()).size() == 5);
```
Comment (houseroad) [inline torch/csrc/jit/passes/onnx/prepare_inplace_ops_for_onnx.cpp:30]: How about the opset < 11, if we set start and end as arguments, not inputs

Comment (shubhambhokare1) [reply torch/csrc/jit/passes/onnx/prepare_inplace_ops_for_onnx.cpp:30]: For this pass the operator is aten::slice, start/end/step would always be inputs. The function
ConvertSliceToIndex is only called on encountering an index_put node.

**Pull request closed without merging**